src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1677 @@
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// You are under no obligation whatsoever to provide any bug fixes, patches, or upgrades to the
+// features, functionality or performance of the source code ("Enhancements") to anyone; however,
+// if you choose to make your Enhancements available either publicly, or directly to Lawrence
+// Berkeley National Laboratory, without imposing a separate written license agreement for such
+// Enhancements, then you hereby grant the following license: a non-exclusive, royalty-free
+// perpetual license to install, use, modify, prepare derivative works, incorporate into other
+// computer software, distribute, and sublicense such enhancements or derivative works thereof,
+// in binary and source code form.
+
+
+// C++ Headers
+#include <HybridEvapCoolingModel.hh>
+
+#include <UtilityRoutines.hh>
+
+#include <string>
+#include <list>
+#include <math.h>
+//#include <windows.h>
+#include <ScheduleManager.hh>
+#include <General.hh>
+#include <CurveManager.hh>
+#include <DataGlobals.hh>
+#include <DataGlobalConstants.hh>
+#include <DataHVACGlobals.hh>
+#include <Psychrometrics.hh>
+#include <DataEnvironment.hh>
+#include <DataZoneEquipment.hh>
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+using namespace std;
+
+namespace EnergyPlus {//***************
+
+	namespace HybridEvapCoolingModel {
+		using namespace std;
+		using DataHVACGlobals::SysTimeElapsed;
+		using DataHVACGlobals::NumOfSysTimeSteps;
+		using DataHVACGlobals::NumOfSysTimeStepsLastZoneTimeStep;
+		using DataHVACGlobals::LimitNumSysSteps;
+		using DataHVACGlobals::TimeStepSys;
+		using DataGlobals::SecInHour;
+		using namespace EnergyPlus::DataEnvironment;
+		using namespace Psychrometrics;
+		using CurveManager::GetCurveIndex;
+		using CurveManager::GetCurveType;
+		using CurveManager::GetCurveMinMaxValues;
+		using CurveManager::CurveValue;
+		using EnergyPlus::CurveManager::GetNormalPoint;
+		using ScheduleManager::GetCurrentScheduleValue;
+		typedef int* MyType;
+		map<string, string> testop;
+#define  DEF_Tdb 0
+#define  DEF_RH 1
+
+#define  TEMP_CURVE 0 
+#define  W_CURVE 1
+#define  POWER_CURVE 2
+		// constructor of Model,
+
+#include <fstream>
+		fstream logfile;
```

Comment:
What are you using a logfile for?  We have an error file and don't want to muddy up the user's run directory.